app/controllers/user_confirmations_controller.rb

```diff
@@ -13,10 +13,10 @@ def create
 
     self.resource = resource_class.send_confirmation_instructions(resource_params)
 
-    if successfully_sent?(resource)
-      set_flash_message(:success, :confirmation_sent) if is_navigational_format?
-    else
-      set_flash_message(:error, :confirmation_not_sent) if is_navigational_format?
+    if successfully_sent?(resource) && is_navigational_format?
```

Comment:
Looks like this could have been:
```ruby
    if is_navigational_format?
      if successfully_sent?(resource)
        set_flash_message(:success, :confirmation_sent)
      else
        set_flash_message(:error, :confirmation_not_sent)
      end
    end
```